Mathlib/Tactic/Linter/Style.lean

```diff
@@ -29,17 +29,18 @@ register_option linter.style.setOption : Bool := {
 namespace Style.setOption
 
 /-- Whether a syntax element is a `set_option` command, tactic or term:
-Return the name of the option being set, if any. -/
-def parse_set_option : Syntax → Option Name
-  -- This handles all four possibilities of `_val`: a string, number, `true` and `false`.
-  | `(command|set_option $name:ident $_val) => some name.getId
-  | `(set_option $name:ident $_val in $_x) => some name.getId
-  | `(tactic|set_option $name:ident $_val in $_x) => some name.getId
+Return the name and value of the option being set, if any.
+-/
+def parse_set_option : Syntax → Option (Name × TSyntax [`str, `num])
```

Comment:
```suggestion
def parse_set_option : Syntax → Option (Syntax × TSyntax [`str, `num])
```
You could return the identifier itself, rather than composing with `getId`, to collect position information.